src/QsCompiler/CommandLineTool/Options.cs

```diff
@@ -117,20 +118,37 @@ public class CompilationOptions : Options
         /// Returns a dictionary with the specified assembly properties as out parameter.
         /// Returns a boolean indicating whether all specified properties were successfully added.
         /// </summary>
-        internal bool ParseAssemblyProperties(out Dictionary<string, string> parsed)
+        internal bool ParseAssemblyProperties(ILogger? logger, [MaybeNullWhen(false)] out Dictionary<string, string> asmProperties)
         {
-            var success = true;
-            parsed = new Dictionary<string, string>();
+            asmProperties = null;
+            var parsed = new Dictionary<string, string>();
```

Comment:
Should an `IEqualityComparer<TKey>` be supplied in the ctor?

Reply:
The default comparer for strings sounds fine here. What would be the reason you had in mind for providing an argument instead of relying on the default behavior of string dictionaries?